src/R/Core/Impl/AST/Operators/Associativity.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Microsoft.R.Core.AST.Operators {
+    // https://en.wikipedia.org/wiki/Operator_associativity
+    /// <summary>
+    /// Associativity. Consider the expression a~b~c. If the operator 
+    /// ~ has left associativity, this expression would be interpreted as (a~b)~c.
+    /// If the operator has right associativity, the expression would be
+    /// interpreted as a~(b~c).
+    /// </summary>
+    public enum Associativity {
```

Comment:
It might, not all edge cases are currently being processed (#1300 is only partial). The reason is (we need to discuss) that design time parsing is often different from run-time parsing. Say, ```x + y <- 1``` may be legal but does user actually want this construct or are they still editing file and forgot to add line break? Ex in HTML browsers are very forgiving but in the editor users prefer strict checks, even when browser allows particular construct.